### PR TITLE
[OneExplorer] Allow rename command only for config files

### PR DESCRIPTION
**Pull request by dayo09**
This commit allows rename command only for config files. By this, base model and products renaming is restricted
and it will be replaced with 'Refactoring' command.

This commit also include some code cleanings.

ONE-vscode-DCO-1.0-Signed-off-by: Dayoung Lee <dayoung.lee@samsung.com>

----

FOR REVIEWERS,

This commit is from discussion https://github.com/Samsung/ONE-vscode/pull/1361
It's decided to enable rename command only for config files,
and support refactoring command for other files.

### CODE REVIEW

File: src/OneExplorer/OneExplorer.ts [hunk 1]
```diff
@@ -206,8 +206,7 @@ class NodeFactory {
     //
     // A 'TreeDataProvider<element>' expects every elements (Node) to be correspond to visible
     // TreeItem, so let's not build hidden nodes.
-    if(attr && attr?.canHide === true && OneTreeDataProvider.didHideExtra === true)
-    {
+    if (attr && attr?.canHide === true && OneTreeDataProvider.didHideExtra === true) {
```
Comment (seanshpark) [inline src/OneExplorer/OneExplorer.ts:209]: Q) can we use `if (attr && attr?.canHide && OneTreeDataProvider.didHideExtra) {` without `=== true)` ?

**Review by seanshpark — APPROVED** [review]
LGTM

File: src/OneExplorer/OneExplorer.ts (hunk 1 above)
Comment (dayo09) [reply src/OneExplorer/OneExplorer.ts:209]: Yes :-D I just explicitly wrote it down as those fields can be 'undefined' or 'false'(either of them is false), but I think it's also clear with the shortened form.

Comment (dayo09) [inline src/OneExplorer/OneExplorer.ts:209]:
```suggestion
    if (attr && attr?.canHide && OneTreeDataProvider.didHideExtra ) {
```

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by jyoungyun — APPROVED** [review]
LGTM

**Pull request merged**